### ISSUE TITLE
CSGMesh does not receive light outside CSG hierarhy

### ISSUE THREAD

**Issue by milkowski** [godotengine/godot#20251]
**Godot version:**
3.1.dev (09b1fdc)

**OS/device including version:**
Windows 10, Radeon RX480

**Issue description:**
CSGMesh instance does no receive lighting if act standalone outside bigger CSG hirerarchy. If connected with other CSG nodes it behaves as expected. 

**Steps to reproduce:**
1. Add CSGMesh with Plane mesh to scene and SpotLight directed to it - no light received.
![image](https://user-images.githubusercontent.com/136565/42899616-e2b42b70-8ac6-11e8-95f4-f5b7a837f3f4.png)
2. Add CSGBox hierarchy and move CSGMesh instance to it - lighting is OK.
![image](https://user-images.githubusercontent.com/136565/42899701-239188cc-8ac7-11e8-98bf-28b140e1fe12.png)

**Minimal reproduction project:**
[scratch_tmp.zip](https://github.com/godotengine/godot/files/2206887/scratch_tmp.zip)

**Comment by jmf** [comment]
@milkowski: The mesh is scaled weirdly. It is scaled by -1 in x and z direction and is scaled to 0 in y direction. When you set the scale to 1 in y direction, the shadow works as expected.
I can't tell why parenting the mesh solves the issue. My guess is that some CSG code changes that as a side-effect.

**Comment by milkowski** [comment]
@jmf Yes the scale factors were negative and making them positive fixes the problem. The open question is it is legal to have scale factor less than zero (formally it would allow mirroring of objects) or equal to zero (ie. to collapse particular dimension and make 2D projection).
Second question is if the problem with lighting in negative scale is due to opposite normals or jus bug.

**Comment by jmf** [comment]
@milkowski: The scale factor can be set to less than zero. That's not the problem.
The problem is, that one scale factor is set to exactly 0, which probably messes up some normal calculations or similar. If you set the scale factor of y to 1, you get a flipped and mirrored mesh and no issues with lighting.

**Comment by akien-mga** [comment]
That's indeed not a bug, but I'll add a configuration warning to prevent setting a scale of zero on any axis.
